flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -440,10 +442,12 @@ private byte[] writeToManifest(long checkpointId) throws IOException {
       return EMPTY_MANIFEST_DATA;
     }
 
+    // Refresh table to get the latest specs map
+    table.refresh();
```

Comment:
is this step necessary? 

this is definitely not needed if we go with the read-only table approach for partition spec.

Reply:
It should be reverted. Will update it.

Reply:
Reverted